### PR TITLE
Tonelli-Shanks interface functions and proof of correctness

### PR DESCRIPTION
**Pull request by jagadishb09**

### CODE REVIEW

File: books/kestrel/number-theory/tonelli-shanks-proof.lisp [hunk 1]
```diff
@@ -0,0 +1,1220 @@
...
+    (defthm lrs-aux-lemma-2
+      (implies (and (natp tt^2^i)
+                    (posp i)
+                    (natp m)
+                    (natp p)
+                    (< 2 p)
+                    (< i m)
+                    (equal (least-repeated-square-aux i tt^2^i m p) 0))
+               (equal (least-repeated-square-aux (- m 1) tt^2^i m p) 0))
+      :hints (("goal"
```
Comment (bendyarm) [inline books/kestrel/number-theory/tonelli-shanks-proof.lisp:131]: Does lowercase "goal" work?  There are a few instances of this.

Comment (jagadishb09) [reply books/kestrel/number-theory/tonelli-shanks-proof.lisp:131]: Yes, it works. I was to do make regression and certify all the files, and I have confirmed that it works with Stephen and Alessandro in one of Friday's meetings.

Comment (bendyarm) [reply books/kestrel/number-theory/tonelli-shanks-proof.lisp:131]: It seems to work with "goal" or "Goal", but not if I take out the hint.  So I guess lowercase "goal" does work!

Comment (jagadishb09) [reply books/kestrel/number-theory/tonelli-shanks-proof.lisp:131]: That's good to know. Thanks.

Comment (MattKaufmann) [reply books/kestrel/number-theory/tonelli-shanks-proof.lisp:131]: Yes, case-insensitivity of the goal-spec is documented behavior.  See
this in :doc hints 

    where [goal-spec] is as described elsewhere (see [goal-spec]) and the

and then see this in :doc goal-spec:

    The matching algorithm is case-insensitive.

**Pull request closed without merging**